histogram/src/config.rs

```diff
@@ -0,0 +1,218 @@
+    pub fn total_bins(&self) -> usize {
+        (self.lower_bin_count + self.upper_bin_count) as usize
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn sizes() {
+        assert_eq!(std::mem::size_of::<Config>(), 32);
+    }
+
+    #[test]
+    // Test that the number of bins matches the expected count
+    fn total_bins() {
+        let config = Config::new(0, 2, 64).unwrap();
+        assert_eq!(config.total_bins(), 252);
+
+        let config = Config::new(0, 7, 64).unwrap();
+        assert_eq!(config.total_bins(), 7424);
+
+        let config = Config::new(0, 14, 64).unwrap();
+        assert_eq!(config.total_bins(), 835_584);
+
+        let config = Config::new(1, 2, 64).unwrap();
+        assert_eq!(config.total_bins(), 248);
+
+        let config = Config::new(8, 2, 64).unwrap();
+        assert_eq!(config.total_bins(), 220);
+
+        let config = Config::new(0, 2, 4).unwrap();
+        assert_eq!(config.total_bins(), 12);
+    }
+
+    #[test]
+    // Test value to index conversions
+    fn value_to_idx() {
+        let config = Config::new(0, 7, 64).unwrap();
+        assert_eq!(config.value_to_index(0), Ok(0));
+        assert_eq!(config.value_to_index(1), Ok(1));
+        assert_eq!(config.value_to_index(256), Ok(256));
+        assert_eq!(config.value_to_index(257), Ok(256));
+        assert_eq!(config.value_to_index(258), Ok(257));
+        assert_eq!(config.value_to_index(512), Ok(384));
+        assert_eq!(config.value_to_index(515), Ok(384));
+        assert_eq!(config.value_to_index(516), Ok(385));
+        assert_eq!(config.value_to_index(1024), Ok(512));
+        assert_eq!(config.value_to_index(1031), Ok(512));
+        assert_eq!(config.value_to_index(1032), Ok(513));
+        assert_eq!(config.value_to_index(u64::MAX - 1), Ok(7423));
+        assert_eq!(config.value_to_index(u64::MAX), Ok(7423));
+    }
+
+    #[test]
+    // Test index to lower bound conversion
+    fn idx_to_lower_bound() {
+        let config = Config::new(0, 7, 64).unwrap();
+        assert_eq!(config.index_to_lower_bound(0), 0);
+        assert_eq!(config.index_to_lower_bound(1), 1);
+        assert_eq!(config.index_to_lower_bound(256), 256);
+        assert_eq!(config.index_to_lower_bound(384), 512);
+        assert_eq!(config.index_to_lower_bound(512), 1024);
+        assert_eq!(
+            config.index_to_lower_bound(7423),
+            18_374_686_479_671_623_680
+        );
+    }
+
+    #[test]
+    // Test index to upper bound conversion
+    fn idx_to_upper_bound() {
+        let config = Config::new(0, 7, 64).unwrap();
+        assert_eq!(config.index_to_upper_bound(0), 0);
+        assert_eq!(config.index_to_upper_bound(1), 1);
+        assert_eq!(config.index_to_upper_bound(256), 257);
+        assert_eq!(config.index_to_upper_bound(384), 515);
+        assert_eq!(config.index_to_upper_bound(512), 1031);
+        assert_eq!(config.index_to_upper_bound(7423), u64::MAX);
+    }
```

Comment:
Shall we add a test for `index_to_range` as well? Can use the same values as these two...

Reply:
Added. Resolving.